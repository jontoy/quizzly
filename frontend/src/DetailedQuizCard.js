import React from "react";
import "./QuizCard.css";
import { Link } from "react-router-dom";

const DetailedQuizCard = ({
  name,
  difficulty,
  numResponses,
  numQuestions,
  activate,
  buttonPrompt,
  reset,
}) => {
  return (
    <div className="QuizCard card p-2 mb-1">
      <h5 className="card-title">{name}</h5>
      <p className="card-text">Difficulty: {difficulty}</p>
      <p className="card-text">
        Answered: {numResponses} of {numQuestions}
      </p>

      <Link to="/quizzes">Back to Quizzes</Link>
      {numQuestions > 0 && (
        <>
          <button onClick={activate}>{buttonPrompt}</button>
          {numResponses > 0 && <button onClick={reset}>Reset Progress?</button>}
        </>
      )}
    </div>
  );
};

export default DetailedQuizCard;
