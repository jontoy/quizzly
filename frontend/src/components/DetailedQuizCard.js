import React from "react";
import "./QuizCard.css";
import { Link } from "react-router-dom";

const DetailedQuizCard = ({
  name,
  difficulty,
  numResponses = 0,
  numQuestions = 0,
  activate,
  buttonPrompt,
  reset,
}) => {
  return (
    <div className="QuizCard card p-2 mb-1 bg-light border-dark">
      <h4 className="card-title">{name}</h4>
      <p className="card-text">Difficulty: {difficulty}</p>
      <p className="card-text">
        Answered: {numResponses} of {numQuestions}
      </p>
      <div className="mb-3">
        <Link className="btn btn-outline-info" to="/quizzes">
          Back to Quizzes
        </Link>
      </div>

      {numQuestions > 0 && (
        <div>
          <button className="btn btn-success" onClick={activate}>
            {buttonPrompt}
          </button>
          {numResponses > 0 && (
            <button className="btn btn-danger ml-1" onClick={reset}>
              Reset Progress?
            </button>
          )}
        </div>
      )}
    </div>
  );
};

export default DetailedQuizCard;
