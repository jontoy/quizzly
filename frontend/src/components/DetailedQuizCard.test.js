import React from "react";
import { render } from "@testing-library/react";
import DetailedQuizCard from "./DetailedQuizCard";
import { MemoryRouter } from "react-router-dom";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <DetailedQuizCard />
    </MemoryRouter>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <DetailedQuizCard />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when there are non-zero questions", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <DetailedQuizCard numQuestions={5} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when there are non-zero questions and answers", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <DetailedQuizCard numQuestions={5} numResponses={3} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
