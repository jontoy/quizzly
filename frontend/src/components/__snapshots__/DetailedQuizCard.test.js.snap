// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="QuizCard card p-2 mb-1 bg-light border-dark"
  >
    <h4
      class="card-title"
    />
    <p
      class="card-text"
    >
      Difficulty: 
    </p>
    <p
      class="card-text"
    >
      Answered: 0 of 0
    </p>
    <div
      class="mb-3"
    >
      <a
        class="btn btn-outline-info"
        href="/quizzes"
      >
        Back to Quizzes
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when there are non-zero questions 1`] = `
<DocumentFragment>
  <div
    class="QuizCard card p-2 mb-1 bg-light border-dark"
  >
    <h4
      class="card-title"
    />
    <p
      class="card-text"
    >
      Difficulty: 
    </p>
    <p
      class="card-text"
    >
      Answered: 0 of 5
    </p>
    <div
      class="mb-3"
    >
      <a
        class="btn btn-outline-info"
        href="/quizzes"
      >
        Back to Quizzes
      </a>
    </div>
    <div>
      <button
        class="btn btn-success"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when there are non-zero questions and answers 1`] = `
<DocumentFragment>
  <div
    class="QuizCard card p-2 mb-1 bg-light border-dark"
  >
    <h4
      class="card-title"
    />
    <p
      class="card-text"
    >
      Difficulty: 
    </p>
    <p
      class="card-text"
    >
      Answered: 3 of 5
    </p>
    <div
      class="mb-3"
    >
      <a
        class="btn btn-outline-info"
        href="/quizzes"
      >
        Back to Quizzes
      </a>
    </div>
    <div>
      <button
        class="btn btn-success"
      />
      <button
        class="btn btn-danger ml-1"
      >
        Reset Progress?
      </button>
    </div>
  </div>
</DocumentFragment>
`;
